src/Compilers/CSharp/Portable/BoundTree/NullabilityRewriter.cs

```diff
@@ -74,18 +74,114 @@ private BoundNode VisitBinaryOperatorBase(BoundBinaryOperatorBase binaryOperator
 
         private T GetUpdatedSymbol<T>(BoundNode expr, T sym) where T : Symbol?
         {
-            if (sym is null) return sym;
+            if (sym is null || _snapshotManager is null) return sym;
 
-            if (_updatedSymbols.TryGetValue((expr, sym), out var updatedSymbol))
+            if (!_snapshotManager.TryGetUpdatedSymbol(expr, sym, out var updatedSymbol))
             {
-                Debug.Assert(updatedSymbol is object);
-                return (T)updatedSymbol;
+                updatedSymbol = sym;
+            }
+            Debug.Assert(updatedSymbol is object);
+
+            switch (updatedSymbol)
+            {
+                case LambdaSymbol lambda:
+                    return (T)remapLambda((BoundLambda)expr, lambda);
+
+                case LocalFunctionSymbol localFunction:
+                    return (T)remapLocalFunction(localFunction);
+
+                case LocalSymbol local:
+                    return (T)remapLocal(local);
+
+                case ParameterSymbol param:
+                    if (_remappedSymbols.TryGetValue(param, out var updatedParam))
+                    {
+                        return (T)updatedParam;
+                    }
+                    break;
             }
 
-            return sym;
+            return (T)updatedSymbol;
```

Comment:
>(T)updatedSymbol [](start = 19, length = 16)

Should we rewrite references to type parameters of local functions?

Reply:
Yes, but this is tracked by [https://github.com/dotnet/roslyn/issues/38922](https://github.com/dotnet/roslyn/issues/38922).

---
In reply to: [329749865](https://github.com/dotnet/roslyn/pull/38769#discussion_r329749865) [](ancestors = 329749865)